src/Java.Interop.Tools.JavaTypeSystem/Adapters/JavaXmlApiExporter.cs

```diff
@@ -0,0 +1,357 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Xml;
+using Java.Interop.Tools.JavaTypeSystem.Models;
+
+namespace Java.Interop.Tools.JavaTypeSystem
+{
+	public static class JavaXmlApiExporter
+	{
+		public static void Save (JavaTypeCollection types, string xmlFile)
+		{
+			using (var writer = XmlWriter.Create (xmlFile, new XmlWriterSettings {
+				Encoding = new UTF8Encoding (false, true),
+				Indent = true,
+				OmitXmlDeclaration = true,
+			}))
+				Save (types, writer);
+		}
+
+		public static void Save (JavaTypeCollection types, XmlWriter writer)
+		{
+			writer.WriteStartElement ("api");
+
+			if (types.Platform.HasValue ())
+				writer.WriteAttributeString ("platform", types.Platform);
```

Comment:
Should this also add/update/replace the `/api/@api-source` attribute? https://github.com/xamarin/monodroid/commit/77fbffa7563a9364cbf8faec3cad6225d72cbe25#diff-517130032c172c39da3469c8897c2e2d11757325ca811847d9f7373c056ad8e9R52

```csharp
writer.WriteAttributeString ("api-source", "JavaTypeSystem");
```

Reply:
It purposely **removes** `"api-source"="class-parse"` because that triggers the `class-parse` -> `jar2xml` conversion.

We can change it to `JavaTypeSystem` and the attribute will be ignored by all our code.  It might be helpful if someone posts an `api.xml` to know that is has gone through java type resolution.